client/components/settings/local_settings_modal_controller.js

```diff
@@ -0,0 +1,26 @@
+class LocalSettingsModalController {
+  constructor (localSettings) {
+    Object.assign(this, { localSettings })
+    this.settings = {}
+  }
+
+  $onInit () {
+    Object.assign(this.settings, this.localSettings.getFromLocalStorage())
```

Comment:
Why not just call `this.loadSettings()`? Why is the `Object.assign` needed here? Notice this only ocours once when the page is starting up.